tests/sdk.test.ts

```diff
@@ -404,4 +406,13 @@ describe("SDK actions tests", () => {
       },
     });
   });
+
+  it("Should return the proper Kernel Key", () => {
+    let options = getPluginOptions({ kernelPublicKey: "" });
```

Comment:
Why don't you change this to either `null` or a dedicated testing placeholder value?

Reply:
I don't understand what you mean. The test is meant to make sure that if an empty string is passed to the parameters it defaults to the `KERNEL_PUBLIC_KEY` which is what is tested here.

Reply:
We shouldn't ever pass an empty string in the first place, thats why I made empty string warnings. We should remove that from happening in the first place!

Reply:
That can't be helped when the variable is not set in the environment and gets defaulted to undefined or an empty string.

Reply:
Then we should check inside of the method and throw for an empty string, or undefined.

Reply:
What I am solving here is:
https://github.com/ubiquity-os-marketplace/daemon-pricing/actions/runs/12135054644/job/33833349702#step:3:6

which is meant to be empty because we want the plugin to use
https://github.com/ubiquity-os/plugin-sdk/blob/a0479e373120b22d30b28510542904b7e4907807/src/constants.ts#L2

so we don't have to set this in every plugin. But `??` will consider only null values (so `""` is valid) when we actually want to exclude all `nullish` values. We do not want to throw if this is missing.

Reply:
When env variable is not defined it defaults to empty string so this particular test makes sense.

@gentlementlegen Why do we allow plugins to set [KERNEL_PUBLIC_KEY](https://github.com/ubiquity-os-marketplace/daemon-pricing/blob/74bd780f45d74ab567df8d35b5abb3ac708fa5bf/src/worker.ts#L28)? Shouldn't we simply always use the default value from the sdk?

Reply:
@rndquu We need to be able to set it because the `beta` bot would use a different private key, and so does a user org (for example when you host your own bot in your own org) so the public matching key is always different. This public key can only work for our `ubiquity-os` bot, in production.